Atomic/syscontainers.py

```diff
@@ -1783,3 +1782,63 @@ def get_image_branch(img):
         repo.prepare_transaction()
         repo.transaction_set_ref(None, get_image_branch(dest), rev)
         repo.commit_transaction(None)
+
+
+    def get_storage_path(self):
+        storage = os.path.sep.join([self._get_system_checkout_path(), ".storage"])
+        return self._canonicalize_location(storage)
+
+
+    def _ensure_storage_for_image(self, repo, img):
+        # Get the rev or raise out of the method
+        try:
+            _, rev = self._resolve_image(repo, img)[0]
+        except (IndexError, TypeError):
+            raise ValueError("Image {} not found".format(img))
+        manifest = self._image_manifest(repo, rev)
+        if manifest is None:
+            raise ValueError("Image `%s` not present" % img)
+        layers = SystemContainers.get_layers_from_manifest(json.loads(manifest))
+
+        storage_path = self.get_storage_path()
+        layers_dir = []
+        for i in layers:
+            layer = SystemContainers._drop_sha256_prefix(i)
+            rootfs = os.path.sep.join([storage_path, layer])
+            layers_dir.append(rootfs)
+            if os.path.exists(rootfs):
+                continue
+            os.makedirs(rootfs)
+            rootfs_fd = None
+            try:
+                rootfs_fd = os.open(rootfs, os.O_DIRECTORY)
+
+                rev_layer = repo.resolve_rev("%s%s" % (OSTREE_OCIIMAGE_PREFIX, layer.replace("sha256:", "")), True)[1]
+                if not rev_layer:
+                    raise ValueError("Layer not found: %s.  Please pull again the image" % layer.replace("sha256:", ""))
+
+                self._checkout_layer(repo, rootfs_fd, rootfs, rev_layer)
+            finally:
+                if rootfs_fd:
+                    os.close(rootfs_fd)
+
+        return layers_dir
+
+
+    def prune_storage(self, repo):
+        storage = self.get_storage_path()
+        if not os.path.exists(storage):
+            return
+        for i in os.listdir(storage):
+            rev_layer = repo.resolve_rev("%s%s" % (OSTREE_OCIIMAGE_PREFIX, i), True)[1]
+            if not rev_layer:
+                shutil.rmtree(os.path.join(storage, i))
+
+
+    def mount_from_storage(self, img, destination):
+        repo = self._get_ostree_repo()
+        if not repo:
+            raise ValueError("OSTree not supported")
```

Comment:
Your call.  You're right the img isn't used looking up the repo, I just wondered if it would be helpful to know which img it was about to look up when things went south.